scripts/mysql_system_tables_fix.sql

```diff
@@ -113,6 +113,14 @@ ALTER TABLE columns_priv
 
 ALTER TABLE func add type enum ('function','aggregate') COLLATE utf8_general_ci NOT NULL;
 
+#
+# Add CHECK for 'name','ret','dl' field.
+#
+
+ALTER TABLE func MODIFY name char(64) COLLATE utf8_bin NOT NULL CHECK (name <> '');
+ALTER TABLE func MODIFY ret tinyint(1) DEFAULT '0' NOT NULL CHECK (ret < 5 and ret >= 0 and ret <> 3);
```

Comment:
ALTER TABLE can handle several modifications at with comma separated modifications. Please use this form, sql over multiple lines with indents is acceptable. Check needs to match other definition.

For alter to succeed, the contraints need to match for all elements in the table.
`DELETE FROM func WHERE name='' OR ret=3 OR NOT ret BETWEEN 0 AND 4 OR dl=''`